R/TerminatorClockTime.R

```diff
@@ -34,20 +35,27 @@ TerminatorClockTime = R6Class("TerminatorClockTime",
         ParamUty$new("stop_time", tags = "required",
           custom_check = custom_check)
       ))
-      super$initialize(param_set = ps, properties = c("single-crit",
-        "multi-crit"))
+      super$initialize(param_set = ps, properties = c("single-crit", "multi-crit"))
+      self$unit = "seconds"
     },
 
     #' @description
     #' Is `TRUE` iff the termination criterion is positive, and `FALSE`
     #' otherwise.
-    #'
-    #' @param archive ([Archive]).
-    #'
     #' @return `logical(1)`.
     is_terminated = function(archive) {
       return(Sys.time() >= self$param_set$values$stop_time)
     }
+  ),
+
+  private = list(
+    .status = function(archive) {
+      max_steps =  as.integer(ceiling(difftime(self$param_set$values$stop_time,
```

Comment:
I know it's cheap but I prefer if things that are to remain constant are just calculated once.